apps/evm/lib/evm/builtin.ex

```diff
@@ -138,14 +134,102 @@ defmodule EVM.Builtin do
           {EVM.Gas.t(), EVM.SubState.t(), EVM.ExecEnv.t(), EVM.VM.output()}
   def run_id(gas, exec_env) do
     data = exec_env.data
-    used_gas = @g_identity_base + @g_identity_byte * MathHelper.bits_to_words(byte_size(data))
+    required_gas = @g_identity_base + @g_identity_byte * MathHelper.bits_to_words(byte_size(data))
 
-    if used_gas <= gas do
-      remaining_gas = gas - used_gas
+    if required_gas <= gas do
+      remaining_gas = gas - required_gas
 
       {remaining_gas, %EVM.SubState{}, exec_env, data}
     else
       {0, %EVM.SubState{}, exec_env, :failed}
     end
   end
+
+  @spec exp_mod(EVM.Gas.t(), EVM.ExecEnv.t()) ::
```

Comment:
Jumping around the yellow paper; it looks like this is the modifier for "Partial payment for an EXP operation". It looks like this can be 0 or 1. I'm struggling to find a high level definition of what this value represents. I'm assuming a value of 1 means this calculation requires more gas then if it's 0 but it's not clear to me yet why these factors are considered.

I'm leaving this comment here more in case someone can succinctly describe why this is part of the specification rather then for any code changes.